net/node/infoUpdate.go

```diff
@@ -148,6 +160,8 @@ func (node *node) reconnect(peer *node) error {
 }
 
 func (node *node) TryConnect() {
+	node.nbrNodes.RLock()
+	defer node.nbrNodes.RUnlock()
```

Comment:
The range of the lock is too large, shrink to only cover the read to the list is more better. Like this:
node.nbrNodes.RLock()
list :=  make([]*node, len(node.nbrNodes.List))
for  _, n := range node.nbrNodes.List {
      list = append(list, n)
}
node.nbrNodes.RUnlock()
for _, n := range list{
     .......
}